hydra-node/json-schemas/api.yaml

```diff
@@ -571,6 +580,25 @@ components:
         timestamp:
           $ref: "api.yaml#/components/schemas/UTCTime"
 
+    PeerHandshakeFailure:
```

Comment:
Should: add information on what failed.

As a user, I would like to know whether my or the other node is outdated - is this data available? Also, in the feature description we called it `PeerProtocolVersionMismatch`, but this name is fine too.